src/main/java/htsjdk/tribble/util/ParsingUtils.java

```diff
@@ -90,35 +82,36 @@ public static InputStream openInputStream(String path)
         return openInputStream(path, null);
     }
 
+    static private final Set<String> URL_SCHEMES = new HashSet<>(Arrays.asList("http", "ftp", "https"));
+
     /**
      * open an input stream from the given path and wrap the raw byte stream with a wrapper if given
      *
      * the wrapper will only be applied to paths that are not http, https, ftp, or file, i.e. any {@link java.nio.file.Path}
      * using a custom filesystem plugin
-     * @param path a uri like string
+     * @param uri a uri like string
      * @param wrapper to wrap the input stream in, may be used to implement caching or prefetching, etc
-     * @return
-     * @throws IOException
+     * @return An inputStream appropriately created from uri and conditionally wrapped with wrapper (only in certain cases
+     * @throws IOException when stream cannot be opened against uri
      */
-    public static InputStream openInputStream(String path, Function<SeekableByteChannel, SeekableByteChannel> wrapper)
+    public static InputStream openInputStream(final String uri, final Function<SeekableByteChannel, SeekableByteChannel> wrapper)
             throws IOException {
 
+        // will be empty string if there's no scheme
+        final String scheme = Optional.ofNullable(IOUtil.getScheme(uri)).orElse("");
         final InputStream inputStream;
-        if (path.startsWith("http:") || path.startsWith("https:") || path.startsWith("ftp:")) {
-            inputStream = getURLHelper(new URL(path)).openInputStream();
-        } else if (IOUtil.hasScheme(path)) {
-            inputStream = new SeekablePathStream(IOUtil.getPath(path), wrapper);
+
+        if (URL_SCHEMES.stream().anyMatch(uri::startsWith)) {
+            inputStream = getURLHelper(new URL(uri)).openInputStream();
+        } else if (scheme.isEmpty()) {
+            File file = new File(uri);
+            inputStream = Files.newInputStream(file.toPath());
```

Comment:
Will this handle the case of named-pipes, as @yfarjoun commented in my previous comment? In addition, is there any way to differentiate the pipes from files? If the scheme is empty and it is a `File`, it feels wrong to me that the method does not return a `SeekablePathStream` and it does if the scheme is "file:".

Reply:
unclear.... @lbergelson ?

Reply:
I assume named pipes can be referred to as either a file or a path, which means that we'd need a different mechanism to check for them.  We don't ever do a check for `isRegularFile` which is I think what we'd need to do.

Reply:
I'm not sure how much we care if we return a SeekablePathStream for a file, vs a simple input stream.  They should have similar performance I hope, and this function only guarantees an input stream.   It is weird though to return different types depending on how you write the path though and it probably has some weird knock on effects. That said, it's not a change from how it was set up before is it?  

Why did you change from the previous order and using hasScheme? It seems like you added a new local variable and some additional logic and then only use it to do what the simpler check accomplished?